packages/core/src/modules/proofs/models/ProofServiceOptions.ts

```diff
@@ -0,0 +1,77 @@
+import type { ConnectionRecord } from '../../connections'
+import type { ProofRequest } from '../formats/indy/models/ProofRequest'
+import type { ProofRecord } from '../repository'
+import type { PresentationPreview } from './PresentationPreview'
+import type { AutoAcceptProof } from './ProofAutoAcceptType'
+import type { ProofProtocolVersion } from './ProofProtocolVersion'
+import type { CreatePresentationFormats, ProposeProofFormats, RequestProofFormats } from './SharedOptions'
+
+// ----- Create Proposal ----- //
+export interface CreateProposalOptions {
+  connectionRecord: ConnectionRecord
+  protocolVersion: ProofProtocolVersion
+  proofFormats: ProposeProofFormats
+  willConfirm?: boolean
+  goalCode?: string
+  comment?: string
+  autoAcceptProof?: AutoAcceptProof
+}
+
+export interface CreateProposalAsResponseOptions {
+  proofRecord: ProofRecord
+  proofFormats: ProposeProofFormats
+  willConfirm?: boolean
+  goalCode?: string
+  comment?: string
+  autoAcceptProof?: AutoAcceptProof
+}
+
+// ----- Out Of Band Proof ----- //
+export interface CreateOutOfBandRequestOptions {
```

Comment:
This doesn't appear to be being used. I believe the `CreateOutOfBandRequest` method should be using this instead of `OutOfBandRequestOptions`?

Reply:
basically, i'm wondering why `OutOfBandRequestOptions` has two types for `proofFormats` (`ProposeProofFormats` and `RequestProofFormats`)? It appears the tests only test with the `RequestProofFormats` type